### PR TITLE
Add more tests for duplicate database references

### LINKED ISSUE #12809: add more tests for adding duplicate db references
Right now file paths aren't really being tested. Also need to add test for reloading a project with references to test if the slashes are in a different direction on mac

### PR DESCRIPTION
**Pull request by kisantia**
This PR fixes #12809. These tests would have caught the problems fixed in #12795.

### CODE REVIEW

File: extensions/sql-database-projects/src/test/project.test.ts [hunk 1]
```diff
@@ -374,21 +375,60 @@ describe('Project: sqlproj content operations', function (): void {
...
+		await project.addDatabaseReference(dacpacReference);
 		should(project.databaseReferences.length).equal(2, 'There should be two database references after adding a reference to test.dacpac');
 		should(project.databaseReferences[1].databaseName).equal('test', 'project.databaseReferences[1].databaseName should be test');
 
 		// try to add reference to test.dacpac again
-		await testUtils.shouldThrowSpecificError(async () => await project.addDatabaseReference({ dacpacFileLocation: Uri.file('test.dacpac'), suppressMissingDependenciesErrors: false }), constants.databaseReferenceAlreadyExists);
+		await testUtils.shouldThrowSpecificError(async () => await project.addDatabaseReference(dacpacReference), constants.databaseReferenceAlreadyExists);
 		should(project.databaseReferences.length).equal(2, 'There should be two database references after trying to add a reference to test.dacpac again');
+
+		const projectReference: IProjectReferenceSettings = {
```
Comment (Benjin) [inline extensions/sql-database-projects/src/test/project.test.ts:397]: This test case seems like it's getting a bit long due to the new case.  Maybe it should be split out?

**Review by Benjin — APPROVED** [review]
